src/views/search/search.jsx

```diff
@@ -60,11 +78,20 @@ class Search extends React.Component {
         }
         const start = pathname.lastIndexOf('/');
         const type = pathname.substring(start + 1, pathname.length);
+        const modeOptions = ['trending', 'popular', ''];
```

Comment:
I don't think this belongs in state, it's a constant. Instead of defining it here, just put something like `const ACCEPTABLE_MODES = ['trending', 'popular', '']` at the top of the file, and then refer to `ACCEPTABLE_MODES` instead of `this.state.acceptableModes`.